st2common/tests/unit/services/test_workflow_rerunning.py

```diff
@@ -0,0 +1,230 @@
+
+        # Register runners.
+        runnersregistrar.register_runners()
+
+        # Register test pack(s).
+        actions_registrar = actionsregistrar.ActionsRegistrar(
+            use_pack_cache=False,
+            fail_on_failure=True
+        )
+
+        for pack in PACKS:
+            actions_registrar.register_from_pack(pack)
+
+    def setUp(self):
+        super(WorkflowExecutionRerunTest, self).setUp()
+
+        # Create temporary directory used by the tests
+        _, self.temp_dir_path = tempfile.mkstemp()
+        os.chmod(self.temp_dir_path, 0o755)  # nosec
+
+        with open(self.temp_dir_path, 'w') as f:
+            f.write('1\n')
+
+    def tearDown(self):
+        if self.temp_dir_path and os.path.exists(self.temp_dir_path):
+            if os.path.isdir(self.temp_dir_path):
+                shutil.rmtree(self.temp_dir_path)
+            else:
+                os.remove(self.temp_dir_path)
+
+    def test_rurun(self):
+        task_route = 0
+        param = {'tempfile': self.temp_dir_path}
+
+        # 1. Fail task1 and workflow
+        # Manually create the liveaction and action execution objects without publishing.
+        wf_meta = self.get_wf_fixture_meta_data(TEST_PACK_PATH, TEST_FIXTURES['workflows'][0])
+        lv_ac_db = lv_db_models.LiveActionDB(action=wf_meta['name'], parameters=param)
+        lv_ac_db, ac_ex_db = ac_svc.create_request(lv_ac_db)
+
+        # Request and pre-process the workflow execution.
+        wf_def = self.get_wf_def(TEST_PACK_PATH, wf_meta)
+        st2_ctx = self.mock_st2_context(ac_ex_db)
+        wf_ex_db = wf_svc.request(wf_def, ac_ex_db, st2_ctx)
+        wf_ex_db = self.prep_wf_ex(wf_ex_db)
+
+        self.run_workflow_step(wf_ex_db, RETURN_TASK, task_route,
+                               status=ac_const.LIVEACTION_STATUS_FAILED,
+                               wf_status=wf_statuses.FAILED)
+
+        # Check workflow status.
+        conductor, wf_ex_db = wf_svc.refresh_conductor(str(wf_ex_db.id))
+        self.assertEqual(conductor.get_workflow_status(), wf_statuses.FAILED)
+        self.assertEqual(wf_ex_db.status, wf_statuses.FAILED)
+
+        # 2. Rerun workflow
+        # Manually create the liveaction and action execution objects without publishing.
+        lv_ac_db = lv_db_models.LiveActionDB(action=wf_meta['name'], parameters=param)
+        lv_ac_db, ac_ex_db = ac_svc.create_request(lv_ac_db)
+
+        # prepare rerun.
+        with open(self.temp_dir_path, 'w') as f:
+            f.write('0')
+
+        st2_ctx = self.mock_st2_context(ac_ex_db, st2_ctx)
+        st2_ctx['workflow_execution_id'] = wf_ex_db.id
+
+        # Request workflow rerun execution.
+        wf_ex_db = wf_svc.request_rerun(ac_ex_db, st2_ctx, OPTIONS)
+        self.assertIsNotNone(wf_ex_db.id)
+        self.assertGreater(wf_ex_db.rev, 0)
+        self.assertEqual(wf_ex_db.action_execution, str(ac_ex_db.id))
+        self.assertEqual(wf_ex_db.status, wf_statuses.RESUMING)
```

Comment:
Can you manually run the workflow execution to completion?

Reply:
Looks like you didn't address my previous comment.  Can you manually run the workflow execution to completion?